src/ceph-rbdnamer

```diff
@@ -3,10 +3,19 @@
 DEV=$1
 NUM=`echo $DEV | sed 's#p.*##g; s#[a-z]##g'`
 POOL=`cat /sys/devices/rbd/$NUM/pool`
+if [ -f /sys/devices/rbd/$NUM/pool_ns ]; then
+    NAMESPACE=`cat /sys/devices/rbd/$NUM/pool_ns`
+else
+    NAMESPACE=""
+fi
 IMAGE=`cat /sys/devices/rbd/$NUM/name`
 SNAP=`cat /sys/devices/rbd/$NUM/current_snap`
-if [ "$SNAP" = "-" ]; then
-	echo -n "$POOL $IMAGE"
-else
-	echo -n "$POOL $IMAGE@$SNAP"
+
+echo -n "$POOL"
+if [ -n "$NAMESPACE" ]; then
+    echo -n "/$NAMESPACE"
```

Comment:
Sorry, I did not see the master PR, where this question should have been asked.

Anyway, just wondering, won't we have a problem if we have a namespace with the name the same as an image in the default namespace, e.g. what will happen if  we have `rbd/test` image, `rbd/test` namespace, and `rbd/test/xyz` image?

As I understand it will try to create `rbd/test` symlink for the first image and `rbd/test/xyz` symlink for the second one, which will fail?

Reply:
Correct.  udev swallows errors, so there would be no symlink for the image that is mapped second.  That's the price of preserving backwards compatibility for non-namespaced images and keeping symlinks to namespaced images look like image specs.

Reply:
I suppose we could prefix namespace directory names with `@` as it is not allowed in the image name. I have no idea if it is worth doing though.

Reply:
We haven't permitted `/` in RBD images names for years and years.

Reply:
But one can't use `/` in a directory name? Or I missed your point...

Reply:
I probably just missed your original point and was responding to the fact that `@` is not allowed in the image name so I was saying `/` is not allowed either. We have the same issue with mapping multiple clusters (e.g. image `foo` on clusters `A` vs `B` attempt to map to the same symlink). Given that the purpose of RBD namespaces was for privilege isolation, I would think that someone mapping the same image name as a namespace name on the same host would be a rare occurrence and not worth special-casing over the handling of the vast majority of use-cases.

Reply:
Exactly.

Reply:
> I suppose we could prefix namespace directory names with `@` as it is not allowed in the image name. I have no idea if it is worth doing though.

If we did that, the symlinks to namespaced images wouldn't look like image specs.